TicketsDemo.Xml/Repositories/XmlTrainRepository.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Data.Repositories;
+
+
+namespace TicketsDemo.Xml.Repositories
+{
+    public class XmlTrainRepository : ITrainRepository
+    {
+        private IXMLReader _reader;
+
+        public XmlTrainRepository(IXMLReader reader)
+        {
+            _reader = reader;
+        }
+
+        public List<Train> GetAllTrains()
+        {
+            List<Train> Trains = _reader.ReadFile<Train>($@"{ AppDomain.CurrentDomain.BaseDirectory}\App_Data\Train.xml", "Trains");
```

Comment:
with XML you can put hierarchical data structure into the single file,like:


<Carriages>
  <Carriage>
    <Id>1</Id>
    <Type>SecondClassSleeping</Type>
    <DefaultPrice>100.00</DefaultPrice>
    <TrainId>1</TrainId>
    <Number>1</Number>
	<Places>
		<Place>
		
		</Place>
	</Places>
  </Carriage>
</Carriages>

Which eliminates the need for looping through the carriage/place files and matching their elements to the train.
Please simplify the code and get rid of the carriages.xml and places.xml